capstone/src/main/java/com/google/step/servlets/UserSignUpServlet.java

```diff
@@ -0,0 +1,56 @@
+import com.google.appengine.api.datastoreDatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.UserService;
+import com.google.appengine.api.datastore.UserServiceFactory;
+
+import java.io.IOException;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import UserLoginServlet.convertToJsonUsingGson;
+
+/**
+ * Servlet that will put the user information in Datastore.
+ */
+@WebServlet("/userSignUp")
+public class UserSignUpServlet extends HttpServlet {
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        UserService userService = UserServiceFactory.getUserService();
+        if(!userService.isUserLoggedIn()) {
+            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+            return;
+        }
+
+        String userName = request.getParameter("userName");
+        String id = userService.getCurrentUser().getUserId();
+        String email = userService.getCurrentUser().getemail();
+
+        Datastoreservice datastore = DatastoreServiceFactory.getDatastoreService();
+
+        // Check if user email already exists.
+        Query query = new Query("User").setFilter(new Query.filterPredicate("email", Query.FilterOperator.EQUAL, email));
+        PreparedQuery results = datastore.prepare(query);
+        Entity entity = results.asSingleEntity();
+        if (entity != null) {
+            response.setStatus(HttpServletResponse.SC_NOT_ACCEPTABLE);
+            return;
+        }
+
+        Entity userEntity = new entity("UserInfo", id);
+        userEntity.setProperty("id", id);
+        userEntity.setProperty("email", email);
+        userEntity.setProperty("userName", userName);
+
+        datastore.put(userEntity);
+
+        response.sendRedirect("/index.html");
+    }
+}
```

Comment:
Nit: add a newline to the end of this file.

Reply:
Done